src/util/fifo.h

```diff
@@ -1,69 +1,116 @@
 #pragma once
 
-#include "pa_ringbuffer.h"
+#include <cstdint>
 
 #include "util/class.h"
 #include "util/math.h"
 
+using ring_buffer_size_t = uint32_t;
+
```

Comment:
This requires some comments about the purpose and limitations 

... Fast, trivial type only single reader and writer ...